app/advancedfilters/taxonFilter.js

```diff
@@ -31,29 +29,56 @@ app.controller('taxonFilter', function($scope, hardCodedDataService,
 
   $scope.apply = function() {
     $scope.$parent.addToQuery('taxonId', getQuery());
-  }
+  };
 
   var getQuery = function() {
-    return _.pluck(_.filter($scope.taxonIds, 'checked'), 'taxId');
-  }
+    return _.pluck(_.filter($scope.taxaIds, 'checked'), 'name');
+  };
+
+  var getTaxaInfo = function (taxaIds, withRedirection) {
+    if (taxaIds.length !== 0) {
+      var taxonomyPromise = taxonomyService.getTaxa(taxaIds);
+      taxonomyPromise.then(function (multipleTaxa) {
+        angular.forEach(multipleTaxa.data.taxonomies, function (taxon) {
+          $scope.taxaMapping[taxon.taxonomyId] = taxon;
+        });
+        angular.forEach(multipleTaxa.data.errors, function (error) {
+          delete $scope.taxaIds[error.requestedId];
+        });
+        if (withRedirection === true) {
```

Comment:
This and the recursion make the code difficult to read. getTaxaInfo(taxIds) should just do _one_ thing: retrieve data and populate the objects. It is then the responsibility of the function which made the initial call to make another call to get redirected ids if any.